### ISSUE TITLE
Slider for Vary instead of radio button

### ISSUE THREAD

**Issue by LeoHeuser** [lllyasviel/Fooocus#1857]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated that my variations are totally off (strong setting) or very close to the input image (light setting). Sometimes I want something in between.

**Describe the idea you'd like**
I would like to replace the two radio buttons with a float-point slider. Where I can set a very light variation (left side = 0.01) or a very strong variation (right side = 1.00). I can also set every point in between like 0.14 or 0.69. So I am also able to explore with the "amount of image" my variation creates.

**Comment by CosmosVoy** [comment]
dont know it this is nesessary, check the developer options:

Forced Overwrite of Denoising Strength of "Vary"
Set as negative number to disable. For developer debugging.

![image](https://github.com/lllyasviel/Fooocus/assets/156086376/6a16def9-d921-4ebe-86ae-24e0966b0928)
![image](https://github.com/lllyasviel/Fooocus/assets/156086376/61291a41-217e-4c9f-b6c9-ffc73fcc4585)

**Comment by drdancm** [comment]
Most people, who are not experts,  have never looked at the Developer Debug Mode. They are not doing any Debugging : )
So I for one think, having sliders rather than radio buttons (all or none) is a wonderful idea.
However, I am glad to learn that I don't have to wait for that improvement, if it ever will happen. So thanks !!

**Comment by RodTDai** [comment]
I think the point here is that should not be a developer setting, with a non-intuitive name, right? It should be an artistic control.
So i think moving it to main UI and doing a normalized floating point slider is a really good move!
Thanks

**Comment by LeoHeuser** [comment]
Yeah that's right. From my point of view it makes more sense to make it more accessible to everyone, so that they can explore the impact of the "vary" quite easily.
I would like to see a slider there instead of two radio buttons which represents the extrema poles of "min" and "max".

Best 👋🏼

**Comment by drdancm** [comment]
If you do implement this suggestion I think it would be even nicer to have a couple of fat dots, or short vertical lines on the slider line showing the location of 0.25, 0.5, 0.75. This would only be graphic marker, so it does not require any programming.

Why is this generally useful? Because it is far easier to remember the location of your set point, with such  marks (like on a ruler).

**Comment by PeterMcLennan** [comment]
I tend to use even values when adjusting for that very reason.  Easy to remember.

**Comment by MindOfMatter** [comment]
Hi

I just created a fork fooocus project version just synced with the original project in main branch
https://github.com/MindOfMatter/Fooocus-MindOfMatter-Edition

I created a pr in my fork project to show what we need to change to enable the feature
(for now it is multiple radio choices, but I will change with slider)

https://github.com/MindOfMatter/Fooocus-MindOfMatter-Edition/pull/8

If you want to test it, you need to replace your fooocus local folder with mine
(according to selected feature branch)

Also my dev branch contains all merged tested pr features in my fork project
https://github.com/MindOfMatter/Fooocus-MindOfMatter-Edition/tree/dev

I hope that it could help you :)

Have a nice day

**Comment by DavidDragonsage** [comment]
I would like to register my support for implementing Vary as a slider with marker lines, located where the two Vary radio buttons are now.

There are far too many useful options hidden in the so-called "Debug" section. As drdancm said, most people are not using Fooocus to try to debug it. Even as a programmer myself, I am not comfortable with going to that section for routine activities.

There is nothing inherently dangerous with adjusting the Denoising Strength. If the image comes out strange, the typical user will just do "Vary" again with a different value. Who are we trying to protect?

Fooocus is currently operating with what seem like paternalistic extremes, and this is a good example. First we have to check "Advanced" below the main canvas (yes, I know that can be switched on by default). Then select the "Advanced" tab, then "Debug" and the "Debug Tools" tab (the default).  Then scroll halfway down the Debug Tools pane that contains a group of fairly random options to find "Forced Overwrite of Denoising Strength of "Vary"".

I really like using Fooocus, but as its grown it has become the opposite of user friendly, in some respects.

**Comment by PeterMcLennan** [comment]
I agree. Denoising strength and CFG scale are two of the most important
variables for manipulating Stable Diffusion.
It would be hugely advantageous for us to have easier access to those items.
